core/src/main/java/com/radixdlt/consensus/liveness/Pacemaker.java

```diff
@@ -218,84 +203,188 @@ private void startRound() {
-                + "This may (but doesn't have to) indicate system clock malfunction. "
-                + "Consider further investigation if this log message appears on a regular basis.");
-      }
-      this.metrics.bft().pacemaker().proposalsWithSubstituteTimestamp().inc();
-      return previousProposerTimestamp;
+  @Override
+  public void preProcessUnsyncedProposalForCurrentOrFutureRound(Proposal proposal) {
+    if (proposal.highQC().getHighestRound().gt(this.highestKnownQcRound)) {
+      this.highestKnownQcRound = proposal.highQC().getHighestRound();
+    }
+    if (proposal.getRound().gt(this.highestReceivedProposalRound)) {
+      this.highestReceivedProposalRound = proposal.getRound();
     }
   }
 
-  private List<RawNotarizedTransaction> getTransactionsForProposal(
-      Round round, QuorumCertificate highestQC) {
-    // If we're at the end of an epoch, we need to generate an empty proposal
-    // - these transactions will get ignored at vertex preparation time anyway
-    // TODO: Remove isEndOfEpoch knowledge from consensus
-    if (highestQC.getProposedHeader().getLedgerHeader().isEndOfEpoch()) {
-      return List.of();
+  @Override
+  public void processProposal(Proposal proposal) {
+    final var proposedVertex = proposal.getVertex().withId(hasher);
+    this.vertexStore.insertVertex(proposedVertex);
+    metrics.bft().successfullyProcessedProposals().inc();
+  }
+
+  @Override
+  public void processBFTUpdate(BFTInsertUpdate update) {
+    log.trace("BFTUpdate: Processing {}", update);
+
+    final var round = update.getHeader().getRound();
+    if (round.lt(this.latestRoundUpdate.getCurrentRound())) {
+      log.trace(
+          "InsertUpdate: Ignoring insert {} for round {}, current round at {}",
+          update,
+          round,
+          this.latestRoundUpdate.getCurrentRound());
+      return;
     }
 
-    final var alreadyExecutedVertices =
-        vertexStore.getPathFromRoot(highestQC.getProposedHeader().getVertexId());
-    final var nextTransactions =
-        proposalGenerator.getTransactionsForProposal(round, alreadyExecutedVertices);
-    this.metrics.bft().pacemaker().proposedTransactions().inc(nextTransactions.size());
-    return nextTransactions;
+    this.maybeLatestInsertUpdate = Optional.of(update);
+
+    final var updateIsInsertionOfTimeoutVertex =
+        this.timeoutVertexIdForCurrentRound
+            .filter(update.getInserted().getVertexHash()::equals)
+            .isPresent();
+
+    switch (this.roundStatus) {
+      case UNDISTURBED -> {
+        // The round is undisturbed but if the timeout has already
+        // occurred (even if it was prolonged), we'll add a timeout flag to
+        // our initial vote and broadcast it to everyone.
+        this.tryToVoteForTheProposalForTheFirstTime(this.scheduledRoundTimeoutHasOccurred);
+      }
+      case PROPOSAL_REJECTED -> {
+        if (updateIsInsertionOfTimeoutVertex) {
```

Comment:
I know this is maybe a semi-big change/rename suggestion, but I wonder if we should call it a  `RejectionVertex` or `DefaultVertex`, as it's not just for timeouts any more? :D

IE this would separate the concept of "timeout vote" (goes to everyone, only after local round timeout), and "vertex" to vote on if there's no proposal or you're rejecting the proposal for some reason.

Not necessarily for this PR - just getting your thoughts 👍

Reply:
`FallbackVertex` perhaps?